pkg/cliplugins/workload/plugin/sync.go

```diff
@@ -161,34 +175,116 @@ func enableSyncerForWorkspace(ctx context.Context, config *rest.Config, workload
 	// Create a service account for the syncer with the necessary permissions. It will
 	// be owned by the workload cluster to ensure cleanup.
 	authResourceName := SyncerAuthResourcePrefix + workloadClusterName
-	sa, err := kubeClient.CoreV1().ServiceAccounts(namespace).Create(ctx, &corev1.ServiceAccount{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:            authResourceName,
-			OwnerReferences: workloadClusterOwnerReferences,
-		},
-	}, metav1.CreateOptions{})
-	if err != nil {
-		return "", fmt.Errorf("failed to create ServiceAccount: %w", err)
+	sa, err := kubeClient.CoreV1().ServiceAccounts(namespace).Get(ctx, authResourceName, metav1.GetOptions{})
+
+	switch {
+	case errors.IsNotFound(err):
+		if sa, err = kubeClient.CoreV1().ServiceAccounts(namespace).Create(ctx, &corev1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:            authResourceName,
+				OwnerReferences: workloadClusterOwnerReferences,
+			},
+		}, metav1.CreateOptions{}); err != nil && !errors.IsAlreadyExists(err) {
+			return "", fmt.Errorf("failed to create ServiceAccount %s|%s/%s: %w", workloadClusterName, namespace, authResourceName, err)
+		}
+	case err == nil:
+		oldData, err := json.Marshal(corev1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				UID:             sa.UID,
+				ResourceVersion: sa.ResourceVersion,
```

Comment:
If you want the UID & ResourceVersion in the json path (as preconditions), you'll need to remove them from oldData